### DISCUSSION TITLE
AlphaTex exporter

### DISCUSSION THREAD

**Discussion by vladsavelyev** [CoderLine/alphaTab#1123] [Ideas]
I couldn't find it in the code base, but is there an exporter into the AlphaTex format? 

It would be amazing to have it, as it would be also a great input to train a language model, given large databases of existing GP files.

**Comment by izszzz** [comment]
AlphaTexExporter was implemented in an older version but has been completely removed.
There is no code history on github or elsewhere, but I found it still on the old alphaTab site.
It is very old and not fully working, but should be able to be used as a reference.
I too hope this feature is restored
![image](https://user-images.githubusercontent.com/46592248/226084678-4d418791-e893-4242-9ad6-361eba52c772.png)

https://www.alphatab.net/docs/guides/breaking-changes-097-098/
https://demo.alphatab.net/alphatex.html#exporter

**Comment by Danielku15** [comment]
Some time ago I prioritized that a Guitar Pro exporter would be more valuable than an alphaTex importer. I think the alphaTex adoption might be rather low, and it also does not support at this point all elements so that a full export of all information would be possible. I added the wish to the roadmap: https://github.com/orgs/CoderLine/projects/12/views/1

Depending on your plans I could check if I could provide a preliminary exporter for your needs. I'm curious to hear what your plans are.

**Comment by izszzz** [comment]
Thank you for your reply.
As you say, I too feel that alphaTex is underutilized in the current alphaTab functionality and is not a priority in the roadmap.
Below are the reasons why I would like to use alphaTex.

- The current data format for music notation and sound representation is not human readable without a dedicated editor.
- We feel that alphaTex is well suited to manage sheet music efficiently and correctly.
- alphaTex is the latest, most popular, and most updated OSS for managing scores in text format.

Specifically how we would like to use alphaTex is as follows.
- Convert MXL files output from OMR such as audiveris to alphaTex.
- Convert gp files to alphaTex.
- Manage alphaTex files like github.
- Archive out-of-copyright scores in alphaTex.

By doing these things, we can turn scores that no one sees anymore due to image or pdf format into easy-to-consume content again.

While alphaTex is very easy to use, it lacks the functionality to accomplish the above goals.
We are currently producing a web application to fulfill the above objectives and checking the demand, so we will not be able to contribute immediately, but we hope to do so someday.

**Comment by vladsavelyev** [comment]
For me, I'm just practicing training large language models. I wrote a version of alphaTex [exporter](https://github.com/vladsavelyev/guitartab/blob/main/gp_to_tex.py#L22) on Python, starting from the [PyGuitarPro](https://github.com/Perlence/PyGuitarPro) GP reader. Though it's lacking some features like bar metadata and staves. 

I converted an archive of 47299 gp3-5 files to alphaTex, and created a [dataset](https://huggingface.co/datasets/vldsavelyev/guitar_tab) on HuggingFace.  I'm currently training a couple of models: a [GPT2](https://wandb.ai/vsavelyev/huggingface/runs/vr8x6xbn/overview?workspace=user-vsavelyev) on the full guitartab dataset, and a [GPT2](https://wandb.ai/vsavelyev/huggingface/runs/6juvli1i/overview?workspace=user-vsavelyev) on just 4-string bass tracks. 

Longer term, I was thinking about coupling alphaTex with the corresponding audio data, and looking into training a model for auto-generating tabs from audio.

Also, in more recent practical terms, I think @izszzz's idea of managing tabs under GitHub is amazing!
